src/Marvin.Web/Code/Validators/ValidationExtentions.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Marvin.Web.Code.Exceptions;
+using Marvin.Web.Code.Extensions;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Marvin.Web.Code.Validators
+{
+    /// <summary>
+    /// Extensions for validating conditions
+    /// </summary>
+    public static class ValidationExtensions
+    {
+        private const string MESSAGE = "At least one validation exception has occurred, see inner exception for details";
+
+        /// <summary>
+        /// Check if a parameter is null
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="validation"><see cref="Validation"/></param>
+        /// <param name="theObject">the object to validate</param>
+        /// <param name="paramName">Name of the parameter</param>
+        /// <returns><see cref="Validation"/></returns>
+        public static IValidation ArgumentNullCheck<T>(this IValidation validation, [NotNullWhen(returnValue: false)] T? theObject, string paramName)
+            where T : class
+        {
+            if (theObject.IsNull())
+            {
+                return validation.AddException(new ArgumentNullException(paramName));
+            }
```

Comment:
Could shorted to a 1 liner here, I'm not sure which formatting I prefer
```csharp
return theObject.IsNull() ? validation.AddException(new ArgumentNullException(paramName)) : validation;
```

or

```csharp
return theObject.IsNull() 
     ? validation.AddException(new ArgumentNullException(paramName)) 
     : validation;
```

possibly even

```csharp
return theObject.IsNull().Match(
    t => validation.AddException(new ArgumentNullException(paramName)),
    f => validation);
```